packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -348,459 +530,512 @@ export class ProxyView extends AbstractUIExtension {
-    private canRenderNode(node: SKNode): boolean {
-        if (this.useSynthesisProxyRendering) {
-            return !this.useSynthesisProxyRendering || (node.properties[ProxyView.RENDER_NODE_AS_PROXY_PROPERTY] as boolean ?? true);
-        }
-        // FIXME: remove if and else once the synthesis proxy properties fully work
-        else {
-            // Not an edge, not a comment/non-explicitly specified region
-            // Don't just use includes("$$") since non-explicitly specified regions may contain nodes
-            return node instanceof SKNode && node.id.charAt(node.id.lastIndexOf("$") - 1) !== "$";
+    /** Returns the proxy rendering for an edge. */
+    private createEdgeProxy(edge: SKEdge, ctx: SKGraphModelRenderer): VNode | undefined {
+        if (edge.opacity <= 0) {
+            // Don't draw an invisible edge
+            return undefined;
         }
-    }
-
-    /** Updates the proxy-view options specified in the {@link RenderOptionsRegistry}. */
-    updateOptions(renderOptionsRegistry: RenderOptionsRegistry): void {
-        this.prevProxyViewEnabled = this.proxyViewEnabled;
-        this.proxyViewEnabled = renderOptionsRegistry.getValue(ProxyViewEnabled);
-
-        const toPercent = 0.01;
-        this.sizePercentage = renderOptionsRegistry.getValue(ProxyViewSize) * toPercent;
-
-        this.clusteringEnabled = renderOptionsRegistry.getValue(ProxyViewClusteringEnabled);
-
-        this.capProxyToParent = renderOptionsRegistry.getValue(ProxyViewCapProxyToParent);
-
-        this.filterUnconnected = renderOptionsRegistry.getValue(ProxyViewFilterUnconnected);
-        this.filterDistant = renderOptionsRegistry.getValue(ProxyViewFilterDistant);
 
-        // Debug
-        this.highlightSelected = renderOptionsRegistry.getValue(ProxyViewHighlightSelected);
+        // Change its id to differ from the original edge
+        /*
+        If ids aren't unique, errors like
+        - "TypeError: Cannot read property 'removeChild' of null"
+        - "DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node."
+        - "TypeError: Cannot read property 'sel' of undefined"
+        can occur
+        */
+        edge.id = this.getProxyId(edge.id);
+        // Clear children to remove label decorators,
+        // use assign() since children is readonly for SKEdges (but not for SKNodes)
+        Object.assign(edge, { children: [] });
+
+        const vnode = ctx.renderProxy(edge);
+        return vnode;
+    }
 
-        const useSynthesisProxyRendering = renderOptionsRegistry.getValue(ProxyViewUseSynthesisProxyRendering);
-        if (this.useSynthesisProxyRendering !== useSynthesisProxyRendering) {
-            // Make sure not to use the wrong renderings if changed
-            this.clearRenderings();
-        }
-        this.useSynthesisProxyRendering = useSynthesisProxyRendering;
+    //////// General helper methods ////////
```

Comment:
are these not also methods for the utils?

Reply:
All of these are either proxy-view specific or cannot be outsourced any more, for example because of caches

Reply:
ok